src/components/Profile.tsx

```diff
@@ -43,15 +45,21 @@ const Profile = (): JSX.Element => {
       : `ProfileBlock__${sectionName}`;
   };
 
-  const saveEditProfile = () => {
-    //this is where we write to the blockchain
+  const saveEditProfile = async () => {
+    if (userId !== undefined && newName !== undefined) {
+      const newProfile = core.activityContent.createProfile({
+        name: newName,
+        icon: profile?.icon,
+      });
+      await saveProfile(userId, newProfile);
+    }
```

Comment:
do we need to wait util we get a response back before setting `setIsEditing(!isEditing)`?

Reply:
not really.